File: Java/2610-arrayTo2D.java
import java.util.*;

class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        List<List<Integer>> lists = new ArrayList<List<Integer>>();
        lists.add(new ArrayList<Integer>());
        Boolean added = false; 
        for (int i=0; i<nums.length; i ++)
        {
            for (List<Integer> list : lists) {
                if (!list.contains(nums[i]) && !added)
                {
                    list.add(nums[i]);
                    added = true;
                }
            }
            if (!added)
            {
                ArrayList<Integer> temp = new ArrayList<Integer>();
                temp.add(nums[i]);
                
                lists.add(temp);
            }  
            added = false;
        }
        return lists;
    }
}

File: C#/2610-arrayTo2D.cs
// You are given an integer array nums. You need to create a 2D array from nums satisfying the following conditions:

// The 2D array should contain only the elements of the array nums.
// Each row in the 2D array contains distinct integers.
// The number of rows in the 2D array should be minimal.
// Return the resulting array. If there are multiple answers, return any of them.

// Note that the 2D array can have a different number of elements on each row.

public class Solution2610 {
    public IList<IList<int>> FindMatrix(int[] nums) {
        IList<IList<int>> lists = new List<IList<int>>(){new List<int>()};
        Boolean added = false; 
        for (int i=0; i<nums.Length; i ++)
        {
            foreach (List<int> list in lists){
                if (!list.Contains(nums[i]) && !added)
                {
                    list.Add(nums[i]);
                    added = true;
                }
            }
            if (!added)
            {
                lists.Add(new List<int>(){nums[i]});
            }  
            added = false;
        }
        return lists;
    }
}